src/librustc/ty/layout.rs

```diff
@@ -769,6 +770,7 @@ pub enum Layout {
     CEnum {
         discr: Integer,
         signed: bool,
+        non_zero: bool,
```

Comment:
```
enum E {
    A = -1,
    B = 1,
}
```
`0` is in the range, but the enum is still non-zero.
(Also, `non_zero: bool` fits into previous padding so it doesn't increase the size of `Layout`)